File: src/main/java/hacker/solution/ListDemo.java
package hacker.solution;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class ListDemo {
//    private List<Integer> insert(List<Integer> list, int position, int y) {
//        list.
//    }
    public static void main(String [] args) {
        Scanner scan = new Scanner(System.in);
        int N = scan.nextInt();
        List<Integer> list = new ArrayList<Integer>();
        while (N-- > 0) {
            int n = scan.nextInt();
            list.add(n);
        }
        int Q = scan.nextInt();
        while (Q-- > 0) {
            String command = scan.next();
            if (command.equals("Insert")) {
                int position = scan.nextInt();
                int y = scan.nextInt();
                list.add(position,y);
            } else {
                int position = scan.nextInt();
                list.remove(position);
            }
        }
        scan.close();
        list.stream().forEach(i -> System.out.print(i+" "));
    }
}

File: src/main/kotlin/hacker/solutionk/ListDemo.kt
package hacker.solutionk

import java.util.*
import java.util.Arrays



object ListDemo {
    @JvmStatic
    fun main(args: Array<String>) {
        val scan = Scanner(System.`in`)
        var N = scan.nextInt()
        var list = ArrayList<Integer>(N)
        for (i in 0 until N) list.add(Integer(scan.nextInt()))
        var Q = scan.nextInt()
        for (i in 0 until Q) {
            val command = scan.next()
            if (command.equals("Insert")) {
                val position = scan.nextInt()
                val y = scan.nextInt()
                list.add(position,Integer(y))
            } else {
                val position = scan.nextInt()
                list.removeAt(position)
            }
        }
        scan.close()
        list.stream().forEach {it -> print(it.toString()+" ")}
    }
}